src/btree/bt_cursor.c

```diff
@@ -896,24 +896,53 @@ __wt_btcur_next_random(WT_CURSOR_BTREE *cbt)
 		WT_ERR(__cursor_func_init(cbt, true));
 		WT_WITH_PAGE_INDEX(
 		    session, ret = __wt_row_random_descent(session, cbt));
+
+		/*
+		 * Random descent may return not-found: the tree might be empty
+		 * or have so many deleted items we didn't find any valid pages.
+		 * We can't return WT_NOTFOUND to the application unless a tree
+		 * is really empty, fallback to skipping through tree pages.
+		 */
+		if (ret == WT_NOTFOUND)
+			goto fallback_walk;
```

Comment:
As far as I can see, we won't have done the calculation of `cbt->next_random_leaf_skip`, so it looks like we'll always find the first record in the tree.  Am I missing something?